### ISSUE TITLE
Nullable, required fields of null value break kubectl client-side validation

### ISSUE THREAD

**Issue by sttts** [kubernetes/kubernetes#85694]
When doing `kubectl edit clusterversions` in an OpenShift cluster with a nullable field being required (compare https://github.com/openshift/api/blob/master/config/v1/0000_00_cluster-version-operator_01_clusterversion.crd.yaml#L153) and of value null, one gets:

```
# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
# clusterversions.config.openshift.io "version" was not valid:
# * : Invalid value: "The edited file failed validation": ValidationError(ClusterVersion.status): missing required field "availableUpdates" in io.openshift.config.v1.ClusterVersion.status
#
apiVersion: config.openshift.io/v1
kind: ClusterVersion
metadata:
  creationTimestamp: "2019-11-27T17:35:03Z"
  generation: 4
  name: version
  resourceVersion: "47107"
  selfLink: /apis/config.openshift.io/v1/clusterversions/version
  uid: 21a49c2c-e43d-4fcc-8af1-363222577b1a
spec:
  channel: stable-4.3
  clusterID: d7300062-09bc-4250-ab0a-777d2970cc3c
  overrides:
  - group: apps/v1
    kind: Deployment
    name: openshift-kube-apiserver-operator
    namespace: kube-apiserver-operator
    unmanaged: true
  upstream: https://api.openshift.com/api/upgrades_info/v1/graph
status:
  availableUpdates: null
  conditions:
  - lastTransitionTime: "2019-11-27T17:53:30Z"
    message: Done applying 0.0.1-2019-11-27-172000
    status: "True"
    type: Available
  ...
```

**Comment by sttts** [comment]
cc @liggitt

**Comment by liggitt** [comment]
is there a way to also fix the faulty kubectl validation in 1.18.x?
